kyo-prelude/shared/src/test/scala/kyo2/kernel/BoundaryTest.scala

```diff
@@ -221,4 +221,78 @@ class BoundaryTest extends Test:
             }
         }
     }
+
+    "seq boundary" - {
+        "empty sequence" in {
+            val result = Boundary[TestEffect, Any](Seq.empty[Int < TestEffect]) { seq =>
+                assert(seq.isEmpty)
+                42
+            }
+            val handled = ContextEffect.handle(Tag[TestEffect], 0, _ => 0)(result)
+            assert(handled.eval == 42)
+        }
+
+        "sequence of pure values" in {
+            val seq = Seq(1, 2, 3)
+            val result = Boundary[TestEffect, Any](seq.map(x => x: Int < TestEffect)) { isolatedSeq =>
+                assert(isolatedSeq.size == 3)
+                isolatedSeq.map(_.eval).sum
```

Comment:
Should we add a test for maintaining the order of the seq?

Reply:
added!